### ISSUE TITLE
Can not launch main.py with --dmtet flag

### ISSUE THREAD

**Issue by SavvaI** [ashawkey/stable-dreamfusion#192]
### Description

I tried to fine-tune the dreamfusion mesh with the dmtet but the script fails miserably with an error
`
[F glutil.cpp:338] eglInitialize() failed
Aborted (core dumped)
`

### Steps to Reproduce

1. Build docker from dockerfile
2. Train dreambooth `main.py -O --text "fantasy character dwarf, full height, realistic, high quality" --workspace trial_dmtet --dmtet --iters 5000 --init_ckpt trial/checkpoints/df.pth`
3. Run dmtet fine-tuning `python3 main.py -O --text "fantasy character dwarf, full height, realistic, high quality" --workspace trial_dmtet --dmtet --iters 5000 --init_ckpt trial/checkpoints/df.pth`
4. Get an error

### Expected Behavior

Finetuned with dmtet mesh

### Environment

Ubuntu 20.04.5 LTS, PyTorch 1.12, CUDA 11.6

**Comment by ashawkey** [comment]
@SavvaI Hi, please check https://github.com/ashawkey/stable-dreamfusion#trouble-shooting.

**Comment by SavvaI** [comment]
> @SavvaI Hi, please check https://github.com/ashawkey/stable-dreamfusion#trouble-shooting.

I am using Nvidia-docker as specified above, so I tried to as suggested in [https://github.com/ashawkey/stable-dreamfusion/issues/131#issuecomment-1414727699](url) but it did not work

**Comment by Walterkd** [comment]
# Solution
Go to ./nerf/renderer.py
Replace  `dr.RasterizeGLContext`  to  `dr.RasterizeCudaContext`
It work well!!
So does the same problem in GET3D and nvdiffrec.

**Comment by SavvaI** [comment]
> # Solution
> Go to ./nerf/renderer.py Replace `dr.RasterizeGLContext` to `dr.RasterizeCudaContext` It work well!! So does the same problem in GET3D and nvdiffrec.

Thank you it helps!

**Comment by SavvaI** [comment]
But after solving the described above issue now it fails while trying to load the mean_density from the checkpoint.
[https://github.com/ashawkey/stable-dreamfusion/blob/3bd3c085afcf97c0a044e77088eee735076228b5/main.py#L148](url)
The problem is, vanilla dreamfusion (without --dmtet) did not save any "mean_density" field in the checkpoint.

**Comment by SavvaI** [comment]
> But after solving the described above issue now it fails while trying to load the mean_density from the checkpoint.
> [https://github.com/ashawkey/stable-dreamfusion/blob/3bd3c085afcf97c0a044e77088eee735076228b5/main.py#L148](url)
> The problem is, vanilla dreamfusion (without --dmtet) did not save any "mean_density" field in the checkpoint.

Ok I get it. The reason is that I had to remove --cuda_ray flag from the `python3 main.py --text "fantasy character dwarf, full height, realistic, high quality" --workspace trial -O --vram_O` because after the recent commit it stopped working and now it gives an error. Before the last commit it worked fine.

**Comment by Walterkd** [comment]
> 

What you saying is remove `--cuda_ray` from this code `parser.add_argument('-O', action='store_true', help="equals --fp16 --cuda_ray --dir_text")` ?

**Comment by SavvaI** [comment]
> What you saying is remove `--cuda_ray` from this code `parser.add_argument('-O', action='store_true', help="equals --fp16 --cuda_ray --dir_text")` ?

Yeah, I was forced to do it, since --cuda_ray stopped working for me after the recent commit (dmtet commit actually)
